taskcluster/adhoc_taskgraph/transforms/signing.py

```diff
@@ -54,38 +56,47 @@ def build_signing_task(config, tasks):
     for task in tasks:
         dep = task["primary-dependency"]
         task["dependencies"] = {"fetch": dep.label}
-        artifact_prefix = task["attributes"].get("artifact_prefix", "public").rstrip('/')
+        artifact_prefix = (
+            task["attributes"].get("artifact_prefix", "public").rstrip("/")
+        )
         if not artifact_prefix.startswith("public"):
-            scopes = task.setdefault('scopes', [])
-            scopes.append(
-                f"queue:get-artifact:{artifact_prefix}/*"
-            )
+            scopes = task.setdefault("scopes", [])
+            scopes.append(f"queue:get-artifact:{artifact_prefix}/*")
         manifest_name = dep.label.replace("fetch-", "")
-        manifest = dep.attributes['manifest']
+        manifest = dep.attributes["manifest"]
         signing_cert = get_signing_cert(manifest, config.params["level"])
         task["worker"]["upstream-artifacts"] = [
             {
                 "taskId": {"task-reference": "<fetch>"},
                 "taskType": "build",
-                "paths": [dep.attributes['fetch-artifact']],
+                "paths": [dep.attributes["fetch-artifact"]],
                 "formats": manifest["signing-formats"],
             }
         ]
-        if "mac-behavior" in manifest:
-            task["worker"]["mac-behavior"] = manifest["mac-behavior"]
-        if "mac-entitlements-url" in manifest:
-            task["worker"]["entitlements-url"] = manifest["mac-entitlements-url"]
-        if "mac-provisioning-profile-url" in manifest:
-            task["worker"]["provisioning-profile-url"] = manifest["mac-provisioning-profile-url"]
-        if "product" in manifest:
-            task["worker"]["product"] = manifest["product"]
+        # Optional keys
+        for task_key, manifest_key in (
+            ("behavior", "mac-behavior"),
+            ("entitlements-url", "mac-entitlements-url"),
+            ("loginitems-entitlements-url", "mac-loginitems-entitlements-url"),
+            (
+                "nativemessaging-entitlements-url",
+                "mac-nativemessaging-entitlements-url",
+            ),
```

Comment:
So this works, but it also means that every time we create a mozillavpn adhoc signing PR, we'll have to populate these, even if they stay static for months/quarters/years.

I mentioned we may want to hardcode this and populate it if our product is `mozillavpn` or whatever. What do you think?